homeassistant/components/cert_expiry/config_flow.py

```diff
@@ -43,60 +43,58 @@ def _prt_in_configuration_exists(self, user_input) -> bool:
         return False
 
     async def _test_connection(self, user_input=None):
-        """Test connection to the server and try to get the certtificate."""
-        host = user_input[CONF_HOST]
+        """Test connection to the server and try to get the certificate."""
         try:
-            await self.hass.async_add_executor_job(
-                get_cert, host, user_input.get(CONF_PORT, DEFAULT_PORT)
+            await get_cert_time_to_expiry(
+                self.hass,
+                user_input[CONF_HOST],
+                user_input.get(CONF_PORT, DEFAULT_PORT),
             )
             return True
-        except socket.gaierror:
-            _LOGGER.error("Host cannot be resolved: %s", host)
-            self._errors[CONF_HOST] = "resolve_failed"
-        except socket.timeout:
-            _LOGGER.error("Timed out connecting to %s", host)
-            self._errors[CONF_HOST] = "connection_timeout"
-        except ssl.CertificateError as err:
-            if "doesn't match" in err.args[0]:
-                _LOGGER.error("Certificate does not match host: %s", host)
-                self._errors[CONF_HOST] = "wrong_host"
-            else:
-                _LOGGER.error("Certificate could not be validated: %s", host)
-                self._errors[CONF_HOST] = "certificate_error"
-        except ssl.SSLError:
-            _LOGGER.error("Certificate could not be validated: %s", host)
-            self._errors[CONF_HOST] = "certificate_error"
+        except TemporaryFailure as err:
+            cause = type(err.__cause__)
+            if cause is socket.gaierror:
+                self._errors[CONF_HOST] = "resolve_failed"
+            elif cause is socket.timeout:
+                self._errors[CONF_HOST] = "connection_timeout"
+            elif cause is ConnectionRefusedError:
+                self._errors[CONF_HOST] = "connection_refused"
+        except ValidationFailure:
+            return True
         return False
 
     async def async_step_user(self, user_input=None):
         """Step when user initializes a integration."""
         self._errors = {}
         if user_input is not None:
-            # set some defaults in case we need to return to the form
             if self._prt_in_configuration_exists(user_input):
```

Comment:
We now prefer to use the unique_id api of config entries to handle this.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-ids

Reply:
Done!